### PR TITLE
resf [TASK][29094 ] Create UI

### PR DESCRIPTION
**Pull request by peartalent**
## Related Tickets
- https://edu-redmine.sun-asterisk.vn/issues/29094

## WHAT
- Create UI top rating
## Evidence (Screenshot or Video)

#
image
# Performance (Optional)

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
General |Code should be self-explanatory. | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Is this function or class too big? If yes, is the function or class having too many responsibilities? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
General | Does the code follow DRY?: Do not Repeat Yourself | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General |Exception handling and cleanup (dispose) resources | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
General | Does the code build and execute correctly? | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/movieintroduction/base/BaseFragment.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.fragment.app.Fragment
+
+abstract class BaseFragment : Fragment() {
+    @get:LayoutRes
+    protected abstract val layoutResource: Int
+    override fun onCreateView(
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/movieintroduction/base/BaseFragment.kt]: phân cách bởi dòng trống

File: app/src/main/java/com/sunasterisk/movieintroduction/base/BaseFragment.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(layoutResource, container, false)
+    }
+}
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/movieintroduction/base/BaseFragment.kt]: lỗi gì đây nhỉ

**Review by aug16vcc — APPROVED** [review]

File: app/src/main/java/com/sunasterisk/movieintroduction/data/source/MovieDataSource.kt [hunk 3]
```diff
@@ -4,9 +4,15 @@ import com.sunasterisk.movieintroduction.data.model.Genres
...
     interface Remote {
         fun getAllGenres(): List<Genres>
         fun getAllMovies(): List<Movie>
     }
-    interface View{}
+
+    interface View {
+        val view: View
+    }
 }
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/movieintroduction/data/source/MovieDataSource.kt]: Sao Movie data source lại chứa cả View? 
Sai mô hình rồi em

File: app/src/main/java/com/sunasterisk/movieintroduction/ui/genres/GenresContact.kt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    interface View {
+        val presenter: Presenter?
+        fun showGenres()
+        fun showError(error: Int)
+    }
+
+    interface Presenter {
+        val view: View
+        fun getALlGenres()
```
Comment (tuannm-1276) [inline app/src/main/java/com/sunasterisk/movieintroduction/ui/genres/GenresContact.kt]:
```suggestion
        fun getAllGenres()
```

File: app/src/main/java/com/sunasterisk/movieintroduction/ui/genres/GenresFragment.kt [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.sunasterisk.movieintroduction.ui.genres
+
+import com.sunasterisk.movieintroduction.R
+import com.sunasterisk.movieintroduction.base.BaseFragment
+class GenresFragment(
+) : BaseFragment(), GenresContact.View {
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/movieintroduction/ui/genres/GenresFragment.kt]:
```suggestion
class GenresFragment : BaseFragment(), GenresContact.View {
```

File: app/src/main/res/layout/item_top_movie.xml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    android:layout_height="wrap_content"
+    android:layout_margin="@dimen/dp_5"
+    android:background="@drawable/bg_item_bottom_movie">
+
+    <ImageView
+        android:id="@+id/imageBackdrop"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_margin="@dimen/dp_5"
+        android:layout_marginStart="92dp"
```
Comment (huytq-2422) [inline app/src/main/res/layout/item_top_movie.xml:16]: hard code

**Pull request merged**